baseapp/baseapp.go

```diff
@@ -297,7 +295,7 @@ func (app *BaseApp) LastBlockHeight() int64 {
 	return app.cms.LastCommitID().Version
 }
 
-func (app *BaseApp) init() error {
+func (app *BaseApp) Init() error {
 	if app.sealed {
 		panic("cannot call initFromMainStore: baseapp already sealed")
```

Comment:
Should this condition really panic? (Panics are meant for _programmer_ errors, not runtime errors.) Or should it return an error?

Reply:
On another look, the app is being sealed in `Init()`, and it seems like the reasoning behind these panics was to prevent the users from modifying an already initialized app.

There is also an exported method called `IsSealed()` that returns true if modifications to the app are prohibited

Reply:
Great points! I agree. I think this should be addressed in a separate PR though since the scope is outside of pruning /snapshot refactor. A few other modifiers would have to be changed.

I created an issue to track this: https://github.com/cosmos/cosmos-sdk/issues/11600

Reply:
I agree that this is consistent with the existing behavior and can be addressed separately 👍